src/tokens/TokenDetails.tsx

```diff
@@ -0,0 +1,325 @@
+export default function TokenDetailsScreen({ route }: Props) {
+  const { tokenId } = route.params
+  const { t } = useTranslation()
+  const token = useTokenInfo(tokenId)
+
+  if (!token) {
+    throw new Error(`token with id ${tokenId} not found`)
+  }
+
+  return (
+    <SafeAreaView style={styles.container} edges={['right', 'bottom', 'left']}>
+      <ScrollView>
+        <View style={styles.titleContainer}>
+          <TokenIcon
+            token={token}
+            viewStyle={styles.tokenImg}
+            testID="TokenDetails/TitleImage"
+            size={IconSize.SMALL}
+          />
+          <Text style={styles.tokenName} testID="TokenDetails/Title">
+            {token.name}
+          </Text>
+        </View>
+        <TokenDisplay
+          amount={token.balance}
+          tokenId={tokenId}
+          style={styles.balance}
+          testID="TokenDetails/Balance"
+        />
+        {!token.isStableCoin && <PriceInfo token={token} />}
+        <Actions token={token} />
+        <Text style={styles.yourBalance}>{t('tokenDetails.yourBalance')}</Text>
+        <TokenBalanceItem token={token} containerStyle={styles.tokenBalanceItem} />
+        {token.infoUrl && (
+          <LearnMore
+            tokenName={token.name}
+            infoUrl={token.infoUrl}
+            analyticsProps={getTokenAnalyticsProps(token)}
+          />
+        )}
+      </ScrollView>
+    </SafeAreaView>
+  )
+}
+
+TokenDetailsScreen.navigationOptions = {
+  ...headerWithBackButton,
+}
+
+function PriceInfo({ token }: { token: TokenBalance }) {
+  const { t } = useTranslation()
+  if (!token.priceUsd) {
+    return (
+      <View style={styles.priceInfo}>
+        <Text style={styles.priceInfoUnavailable}>{t('tokenDetails.priceUnavailable')}</Text>
+      </View>
+    )
+  }
+
+  if (!token.historicalPricesUsd || !isHistoricalPriceUpdated(token)) {
+    return null
+  }
+
+  return (
+    <View style={styles.priceInfo}>
+      <PercentageIndicator
+        comparedValue={token.historicalPricesUsd.lastDay.price}
+        currentValue={token.priceUsd!}
+        percentageTextStyle={styles.priceInfoText}
+        UpIcon={DataUp}
+        DownIcon={DataDown}
+        NoChangeIcon={DataUp}
+        suffixText={t('tokenDetails.priceDeltaSuffix') ?? undefined}
+        suffixTextStyle={typeScale.bodySmall}
+        testID="TokenDetails/PriceDelta"
+      />
+    </View>
+  )
+}
+
+function Actions({ token }: { token: TokenBalance }) {
+  const { t } = useTranslation()
+  const sendableTokens = useSendableTokens()
+  const swappableTokens = useSwappableTokens()
+  const cashInTokens = useCashInTokens()
+  const cashOutTokens = useCashOutTokens()
+  const isSwapEnabled = useSelector(isAppSwapsEnabledSelector)
+
+  const actions = [
+    {
+      name: TokenDetailsActionName.Send,
+      text: t('tokenDetails.actions.send'),
+      iconComponent: QuickActionsSend,
+      onPress: () => {
+        navigate(Screens.Send, { defaultTokenOverride: token.address! })
```

Comment:
I recently saw a refactor [here](https://github.com/valora-inc/wallet/pull/4242) by @MuckT that changes the meaning of defaultTokenOverride from address to tokenId, but the type is still string. Some faint alarm bell went off when I saw that, and I'm realizing now that I was worried that one of you could merge the changes without seeing this change and we'd end up with a bug. @MuckT what do you think should be done to prevent this? Maybe the `defaultTokenOverride` parameter name could be changed (in Tom's PR) so we get a compilation error if someone makes this mistake

Reply:
Happy to change it from `defaultTokenOverride` to `defaultTokenIdOverride` in #4242.

Reply:
Left a TODO, so this can be addressed depending on which one merges first

Reply:
I think Tom's suggestion of changing the variable name is safer, since it doesn't depend on merge order-- in either case, a build error should take place if one is merged without the other taking the changes into account yet